### ISSUE TITLE
bootstrap.py is affected by the pypi HTTPS change

### ISSUE THREAD

**Issue by adamcheasley** [collective/buildout.python#68]
From what I can tell, bootstrap.py relies on ez_setup which has a hardcoded URL to http:// for pypi. I cannot get a fresh buildout.python to successfully run bootstrap.py

**Comment by mpeeters** [comment]
Pypi drop support of http on last thursday, here is a example of the problem encountered by current bootstrap

```
$ wget http://pypi.python.org/packages/2.4/s/setuptools/setuptools-0.6c11-py2.4.egg                                                                                                             17:27
--2017-10-30 17:27:20--  http://pypi.python.org/packages/2.4/s/setuptools/setuptools-0.6c11-py2.4.egg
Resolving pypi.python.org... 151.101.36.223
Connecting to pypi.python.org|151.101.36.223|:80... connected.
HTTP request sent, awaiting response... 403 SSL is required
2017-10-30 17:27:20 ERROR 403: SSL is required.
```

**Comment by adamcheasley** [comment]
Indeed. This issue can be fixed at the buildout level, which has already been committed to this repo. The issue we're seeing is when trying to run bootstrap, which has hard coded http URLs in a few places.

**Comment by hvelarde** [comment]
@mauritsvanrees @idgserpro FYI

**Comment by aclark4life** [comment]
Cool, good time to talk about #67. If ``bootstrap.py`` continues to be used, everyone should agree on the ``setuptools`` and ``zc.buildout`` versions used (I don't think I've used ``setuptools-0.6c11-py2.4.egg`` in years). I've had the most consistent results with ``buildout.python`` when using ``setuptools==1.4.2`` and ``zc.buildout==1.7.1`` (i.e. the last Py2.4 compatible versions of each.)

**Comment by idgserpro** [comment]
Although not the "correct" way (because there's a bootstrap for zc.buildout 1.x and 2.x), we've been using https://bootstrap.pypa.io/bootstrap-buildout.py for zc.buildout 1.7.x as well as 2.x without problems.

**Comment by adamcheasley** [comment]
FWIW, we found a solution that worked:
- pip install setuptools==1.4.2
- pip install zc.buildout==1.4.4

Now using the system wide buildout, we can run the main buildout config.

**Comment by aclark4life** [comment]
@adamcheasley Please confirm ``zc.buildout==1.7.1`` works too so we can all agree on a version for #67

**Comment by adamcheasley** [comment]
`zc.buildout==1.7.1` seemed to work fine, once I'd changed the version pin in src/base.cfg.
FYI, we're now seeing further errors due to hardcoded http URLs:
https://github.com/collective/buildout.python/blob/master/src/python24.cfg#L91

```
Installing virtualenv172.
Downloading http://pypi.python.org/packages/source/v/virtualenv/virtualenv-1.7.2.tar.gz
While:
  Installing virtualenv172.
Error: Error downloading extends for URL http://pypi.python.org/packages/source/v/virtualenv/virtualenv-1.7.2.tar.gz: (403, 'SSL is required')
```

**Comment by mauritsvanrees** [comment]
@adamcheasley I just fixed the hardcoded http URLs in commit 3155ecd1d5eb05a2f4e54e02706a64751c9c38f7.

**Comment by adamcheasley** [comment]
@mauritsvanrees Thank you!
